iOS/Airbnb/Airbnb/View/DataSource/SearchViewDataSource.swift

```diff
@@ -0,0 +1,117 @@
+        switch homeData[section] {
+        case let .advertising(items):
+            return items.count
+        case let .destination(items):
+            return items.count
+        case let .livingSpot(items):
+            return items.count
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        switch homeData[indexPath.section] {
+        case let .advertising(items):
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: AdvertisingCell.identifier,
+                                                                for: indexPath) as? AdvertisingCell else {
+                return UICollectionViewCell()
+            }
+            cell.configure(image: items[indexPath.item].image)
+            
+            return cell
+        case let .destination(items):
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: NearbyDestinationCell.identifier,
+                                                                for: indexPath) as? NearbyDestinationCell else {
+                return UICollectionViewCell()
+            }
+            let item = items[indexPath.item]
+            cell.configure(image: item.image, title: item.title, distance: item.distance)
+            
+            return cell
+        case let .livingSpot(items):
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: LivingSpotCell.identifier,
+                                                                for: indexPath) as? LivingSpotCell else {
+                return UICollectionViewCell()
+            }
+            let item = items[indexPath.item]
+            cell.configure(image: item.image, title: item.title)
+            
+            return cell
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+        switch kind {
+        case UICollectionView.elementKindSectionHeader:
+            guard let headerView = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: CollectionHeaderView.identifier, for: indexPath) as? CollectionHeaderView else {
+                return UICollectionReusableView()
+            }
+            let sectionType = Section.allCases[indexPath.section]
+            
+            switch sectionType {
+            case .destination:
+                headerView.setHeaderText(text: "가까운 여행지 둘러보기")
+                headerView.setHeaderFontSize(size: 22)
+            case .livingSpot:
+                headerView.setHeaderText(text: "어디에서나,\n여행은 살아보는거야!")
+                headerView.setHeaderFontSize(size: 22)
+            case .otherwise:
+                headerView.setHeaderText(text: "")
+            }
```

Comment:
이 부분에 대해 고민을 많이해보고 위와 같이 시도해보았습니다.
하지만 제대로 동작하지않아 시간을 많이 잡아먹었네요..ㅠㅠ 시행착오는 아래와 같습니다...
indexPath.section의 데이터는 Int값으로 순차적으로는 1 -> 2 이렇게 들어왔습니다.
하여 아래와 같이 enum으로 좀 더 의미가 잘 전달될 수 있도록 구현하였습니다.
```Swift
enum Section: Int, CaseIterable {
    case destination = 0
    case livingSpot = 1
    case otherwise = 2
}
``` 
이렇게 구현하니 해당되는 case에 맞춰서 들어갈 줄 알았는데 잘못된 HeaderText로 넘어가져서 아래 화면처럼 제대로 된 데이터가 들어가지 않았습니다. 하여 기존대로 진행을 해야할 지 아니면 다른 방법이 있는지 고민이 됩니다...!
<img src="https://user-images.githubusercontent.com/92699723/171124880-1277c68a-f8b3-483b-94db-54f5190478c9.png"  width="400" height="400"/>

Reply:
제가 직접 빌드를 해볼 수 없는 상황이여서, 문제가 어디에서 발생하는지 확인할 수 없는게 너무 안타깝네요.
다만 제가 느낀 점은 문제가 발생할 수 있는공간이 enum 내부는 존재하지 않는 거 같습니다.
해당 코드 부분에서 정확하게 indexPath.section값이 전달하는지, Int -> Section으로 변경할때 원하는 값으로 변경하는지 체크해보시겠어요?

Reply:
indexPath.section 값은 원하는 값으로 잘 들어오는것을 확인하여 더 이상했던 것 같습니다.. 
시도를 많이 해보았는데 아무래도 enum타입에 Int로 값을 지정하면 순차적으로 생성되는 방식으로만 채택하는 것을 좀 깨달은 것 같습니다.
수정한 내용은 아래와 같습니다.
```Swift
let sectionType = SectionList.allCases[indexPath.section]
            switch sectionType {
            case .otherwise:
                headerView.setHeaderText(text: "")
            case .destination:
                headerView.setHeaderText(text: "가까운 여행지 둘러보기")
                headerView.setHeaderFontSize(size: 22)
            case .livingSpot:
                headerView.setHeaderText(text: "어디에서나,\n여행은 살아보는거야!")
                headerView.setHeaderFontSize(size: 22)
            }
enum SectionList: Int, CaseIterable {
    case otherwise = 0
    case destination
    case livingSpot
}
```